### PR TITLE
Re-add .buildkite/pipeline.yml

### PR DESCRIPTION
**Pull request by Anviking**
Attempt to revert input-output-hk/cardano-wallet#3623 if it works with new buildkite agents.

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    agents:
+      system: x86_64-linux
+      queue: benchmark
+
+  - label: 'HLint'
+    depends_on: nix
+    command: 'nix develop --command bash -c "echo +++ HLint ; hlint lib"'
+    agents:
+      system: x86_64-linux
+      queue: benchmark
```
Comment (Anviking) [inline .buildkite/pipeline.yml:54]: Let's remove these some day after we've merged #3621

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      - './nix/regenerate.sh'
+    agents:
+      system: x86_64-linux
+      queue: benchmark
+
+  - label: 'Check Cabal Configure (Haskell.nix shellFor)'
+    depends_on: nix
+    command: 'nix develop --command scripts/buildkite/check-haskell-nix-cabal.sh'
+    agents:
+      system: x86_64-linux
```
Comment (piotr-iohk) [inline .buildkite/pipeline.yml:39]:
```suggestion
      system: x86_64-linux
      queue: benchmark
```

**Review by piotr-iohk — APPROVED** [review]
:+1:

**Comment by Anviking** [comment]
Thanks for the fix (and idea to begin with) @piotr-iohk 🙏

**Pull request merged**